stdlib/in_channel.mli

```diff
@@ -0,0 +1,134 @@
+(**************************************************************************)
+(*                                                                        *)
+(*                                 OCaml                                  *)
+(*                                                                        *)
+(*             Xavier Leroy, projet Cristal, INRIA Rocquencourt           *)
+(*                                                                        *)
+(*   Copyright 2019 Institut National de Recherche en Informatique et     *)
+(*     en Automatique.                                                    *)
+(*                                                                        *)
+(*   All rights reserved.  This file is distributed under the terms of    *)
+(*   the GNU Lesser General Public License version 2.1, with the          *)
+(*   special exception on linking described in the file LICENSE.          *)
+(*                                                                        *)
+(**************************************************************************)
+
+
+(** Input channels.
+
+    @since 4.10 *)
+
+type t = in_channel
+
+val stdin : t
+(** Standard output. *)
+
+(** {2 General input functions} *)
+
+type 'a open_function =
+  ?create:bool -> (* default: false *)
+  ?binary:bool -> (* default: false *)
+  'a
+
+val create : t open_function
+(** TBA *)
+
+val with_file : (string -> (t -> 'a) -> 'a) open_function
```

Comment:
I add it here, so I don't forget to re-check later.

If we accept the `with_file` functions (and I really welcome their introduction) we should mention that the user has to be careful and avoid capturing the environment, for example leaking no-longer open channels.